spring-cloud-config-client/src/test/java/org/springframework/cloud/config/client/ConfigServerConfigDataLocationResolverTests.java

```diff
@@ -139,10 +142,35 @@ void uriWithParamsParsesProperties() {
 	}
 
 	@Test
-	void urisInLocationOverridesProperty() {
-		String locationUri = "http://actualuri1,http://actualuri2";
-		ConfigServerConfigDataResource resource = testUri("http://shouldbeoverridden", locationUri);
-		assertThat(resource.getProperties().getUri()).containsExactly(locationUri.split(","));
+	void useExistingConfigClientPropertiesInBootstrapContext() {
+		ConfigurableBootstrapContext bootstrapContext = mock(ConfigurableBootstrapContext.class);
+		when(bootstrapContext.isRegistered(eq(ConfigClientProperties.class))).thenReturn(true);
+		ConfigClientProperties configClientProperties = new ConfigClientProperties();
+		configClientProperties.setUri(new String[] { "http://myuri" });
+		when(bootstrapContext.get(eq(ConfigClientProperties.class))).thenReturn(configClientProperties);
+		when(context.getBootstrapContext()).thenReturn(bootstrapContext);
+		List<ConfigServerConfigDataResource> resources = this.resolver.resolveProfileSpecific(context,
+				ConfigDataLocation.of("configserver:http://locationuri"), mock(Profiles.class));
+		assertThat(resources).hasSize(1);
+		verify(bootstrapContext, times(1)).get(eq(ConfigClientProperties.class));
+		ConfigServerConfigDataResource resource = resources.get(0);
+		assertThat(resource.getProperties().getUri()).isEqualTo(new String[] { "http://locationuri" });
```

Comment:
Should there be a test that is just `configserver:` that gets the uri from the existing properties?